js_modules/dagster-ui/packages/ui-core/src/runs/useRunsForTimeline.tsx

```diff
@@ -317,16 +327,71 @@ export const useRunsForTimeline = ({
         startTime: run.startTime * 1000,
         endTime: run.endTime ? run.endTime * 1000 : now,
       };
-    });
+      if (!jobInfo[runJobKey] && run.repositoryOrigin) {
+        const pipelineName = run.pipelineName;
+        const isAdHoc = isHiddenAssetGroupJob(pipelineName);
 
-    return map;
-  }, [ongoingRunsData, completedRunsData, runsNotCapturedByUpdateBuckets, start, end]);
+        const repoAddress = buildRepoAddress(
+          run.repositoryOrigin!.repositoryName,
+          run.repositoryOrigin!.repositoryLocationName,
+        );
+        jobInfo[runJobKey] = {
+          repoAddress,
+          isAdHoc,
+          pipelineName,
+        };
+      }
+    }
 
-  const jobsWithRuns: TimelineJob[] = useMemo(() => {
-    if (!workspaceOrError || workspaceOrError.__typename === 'PythonError') {
-      return [];
+    // fetch all the runs in the given range
+    completedRuns.forEach(saveRunInfo);
+    ongoingRunsData?.forEach(saveRunInfo);
+    const current = {jobInfo, runsByJobKey: map};
+    previousRunsByJobKey.current = current;
+    return current;
+  }, [completedRuns, ongoingRunsData, loading]);
+
+  const jobsWithCompletedRunsAndOngoingRuns = useMemo(() => {
+    const jobs: Record<string, TimelineJob> = {};
+    if (!Object.keys(runsByJobKey).length) {
+      return jobs;
     }
 
+    Object.entries(runsByJobKey).forEach(([jobKey, jobRunsInfo]) => {
+      const runs = Object.values(jobRunsInfo);
+      const info = jobInfo[jobKey];
+      if (!info) {
+        return;
+      }
+
+      const {pipelineName, isAdHoc, repoAddress} = info;
+
+      jobs[jobKey] = {
+        key: jobKey,
+        jobName: isAdHoc ? 'Ad hoc materializations' : pipelineName,
+        jobType: isAdHoc ? 'asset' : 'job',
+        repoAddress,
+        path: workspacePipelinePath({
+          repoName: repoAddress.name,
+          repoLocation: repoAddress.location,
+          pipelineName,
+          isJob: true,
```

Comment:
Pretty sure this is always true. I grepped for `resolve_isJob` in OSS and Internal, [this is the only implementation of it](https://github.com/dagster-io/dagster/blob/f6c842b29e514c0a5354e5dceabc4b7bc23a23e3/python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py#L904)